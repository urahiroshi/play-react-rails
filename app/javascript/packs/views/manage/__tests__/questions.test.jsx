import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Questions from '../questions'

describe('when questions are empty', () => {
  it('should show empty tbody', () => {
    const wrapper = shallow(<Questions questions={[]} />)
    expect(wrapper.matchesElement(<div />)).toBe(true)
  })
})

describe('when questions are not empty', () => {
  const questions = [
    {
      id: 1,
      content: 'hogehoge',
      answer: 'hoge',
    }, {
      id: 2,
      content: 'fugafuga',
      answer: 'fuga',
    },
  ]

  it('should show question rows', () => {
    const wrapper = shallow(<Questions questions={questions} />)
    expect(wrapper.find('tbody').children().length).toBe(questions.length)
  })

  it('should show snapshot', () => {
    const wrapper = shallow(<Questions questions={questions} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
