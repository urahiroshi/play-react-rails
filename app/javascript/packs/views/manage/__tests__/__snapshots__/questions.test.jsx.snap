// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when questions are not empty should show snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        className="col-md-1"
      >
        Index
      </th>
      <th
        className="col-md-6"
      >
        Content
      </th>
      <th
        className="col-md-3"
      >
        Answer
      </th>
      <th
        className="col-md-2"
      />
    </tr>
  </thead>
  <tbody>
    <Connect(QuestionRow)
      order={1}
      question={
        Object {
          "answer": "hoge",
          "content": "hogehoge",
          "id": 1,
        }
      }
    />
    <Connect(QuestionRow)
      order={2}
      question={
        Object {
          "answer": "fuga",
          "content": "fugafuga",
          "id": 2,
        }
      }
    />
  </tbody>
</table>
`;
